enhancements/monitoring/alerts-ui-management.md

```diff
@@ -0,0 +1,642 @@
+
+**Could-Have Features:**
+- Notifications (alertmanager receivers)
+- PromQL expression autocompletion and graph
+- "Save as draft" wizard
+- Alert rule history
+- Acknowledge alert
+- Filter by triggered date/time
+- Column management
+- Additional alert action items (View logs, Troubleshoot, etc.)
+- Generate a summary report
+- Generate a dashboard
+- Manage groups
+
+**Won't-Have Features:**
+- Alertmanager sub-tab
+
+### API Endpoints
+
+Base path: `/api/v1/alerting`
+
+Rule identity
+
+- A rule is uniquely identified by the tuple: `(namespace, prometheusrule, ruleName, specHash)`.
+- Canonical opaque identifier used below as `ruleId`:
+  - `/namespaces/{namespace}/prometheusrules/{prometheusrule}/rules/{ruleName}/specHash/{specHash}`.
+    - `specHash` is server‑generated from the rule spec to ensure uniqueness and stability. It is computed as SHA‑256 (hex, lowercase) of the normalized fields:
+      - `expr`: trimmed with consecutive whitespace collapsed
+      - `for`: normalized to seconds
+      - `labels`: Validate labels `key` and `value` meet Prometheus syntax and character rules; drop empty values; sort by key ascending; join as `key=value` lines separated by `\n`
+      - Concatenate the three parts with `\n---\n` separators, then hash the UTF‑8 bytes
+    - Clients must treat `specHash` as opaque (do not construct it). When `expr`/`for`/labels change, the server recomputes `specHash` and returns an updated `ruleId`.
+
+**Alerts (instances)**
+
+| Method | Path                      | Description                                                                                         | Notes |
+|--------|---------------------------|-----------------------------------------------------------------------------------------------------|-------|
+| GET    | `/alerts`                 | List alert instances (Pending / Firing / Silenced) with post‑relabel labels. Supports the filter parameters listed below. | **Required** — merges `AlertRelabelConfig` with platform alerts. |
+
+Filter parameters for `/alerts`:
+
+- `name`
+- `group`
+- `component`
+- `severity`
+- `state` (one of: `firing`, `pending`, `silenced`)
+- `source` (platform, user-defined)
+- `notification_receiver`
+- `triggered_since`
+- Other alert labels (arbitrary `labelKey=labelValue` pairs)
+
+**Rules (definitions)**
+
+| Method | Path                                                  | Description                                                                                              | Notes |
+|--------|-------------------------------------------------------|----------------------------------------------------------------------------------------------------------|-------|
+| GET    | `/rules`                           | List all alert rule definitions **(platform and user)**, including `disabled` flags and relabel context. Supports the filter parameters listed below. | **Required** — merges `AlertRelabelConfig` with platform alerts.|
```

Comment:
Hi @simonpasquier @jgbernalp , @jan--f I would like to drop the following APIs:
- `GET /rules`
-  `GET /rules/labels`
-  `GET /rules/{ruleId}` 

 in favor of persisting an aggregated relabeled alerting rules list in a ConfigMap.

I believe this would enable the UI to use the plain k8s API in the single cluster and in ACM it can be tracked by the ACM searched and used by the UI.

Please let me know if you agree.

Reply:
The goal was to allow the UI to use the same APIs for single or multi cluster contexts, by having the backend abstract away the complexities and aggregations. Also we might loose the independent access control for rules if all of them are in a single ConfigMap, for cluster with multiple rules loading a single ConfigMap will affect performance.

Reply:
We need to sync with the ACM team in terms how this is currently being managed, this does not answer what happens when a user wants to manage a specific alert, how do we control access? Is that a requirement?. Currently the alerts have the information of the cluster they come from as a label.